### PR TITLE
feat(go): resource overrides

### PR DESCRIPTION
**Pull request by iliapolo**
Golang version of [`resource-overrides`](https://github.com/aws-samples/aws-cdk-examples/tree/master/typescript/resource-overrides). 

Still not working. See review comments.

Related to https://github.com/aws/jsii/issues/2676

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: go/resource-overrides/resource-overrides.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	bucket := awss3.NewBucket(stack, jsii.String("MyBucket"), &awss3.BucketProps{
+			Versioned: jsii.Bool(true),
+			Encryption: awss3.BucketEncryption_KMS_MANAGED,
+	});
+
+	defaultChild := bucket.Node().DefaultChild()
+	bucketResource2, ok := defaultChild.(awss3.CfnBucket)
+
+	if !ok {
```
Comment (iliapolo) [inline go/resource-overrides/resource-overrides.go:36]: The casting here fails. I wonder if i'm missing something and there is another way to do it.

Possibly related to https://github.com/aws/jsii/blob/main/packages/@jsii/go-runtime-test/project/compliance_test.go#L749

?

Comment (iliapolo) [reply go/resource-overrides/resource-overrides.go:36]: cc @MrArnoldPalmer @RomainMuller @eladb

Comment (eladb) [reply go/resource-overrides/resource-overrides.go:36]: What is the type you are getting?

**Comment by indrora** [comment]
Thank you for your contribution. Unfortunately, this PR is very old, has had no movement, and predates the launch of CDK v2. 

I'm going to close this PR for now. If you update it to use the CDKv2, please open a new PR.

**Pull request closed without merging**